spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientConfigReloadAutoConfiguration.java

```diff
@@ -55,7 +58,7 @@
  * @author Ryan Baxter
  */
 @Configuration(proxyBeanMethods = false)
-@ConditionalOnKubernetesAndConfigEnabled
```

Comment:
`ConditionalOnKubernetesAndConfigEnabled` is saying that we only need to take this configuration into account if config-maps support is enabled and we are inside k8s. At the same time it contains beans that are secret based. So, it was probably left by accident here, when support for secrets was added.

Besides, beans in this configuration have for example such annotations:

```
@ConditionalOnBean(KubernetesClientConfigMapPropertySourceLocator.class)
```

And a bean of type `KubernetesClientConfigMapPropertySourceLocator` is only available if config-maps support is present, so it kind of duplicates the point of `ConditionalOnKubernetesAndConfigEnabled`